tests/testthat/test_ggetbout.R

```diff
@@ -60,32 +60,87 @@ test_that("g.getbout produces expected output", {
+  # in this way, we can predict the time in bouts that should be detected
+  # -------------
+  dummybout = function(totaldur_min = 60,
+                       ws3 = 5,
+                       boutdur = 10,
+                       boutdur_detect = 10,
+                       boutcriter = 0.8,
+                       nBouts = 1) {
+    x = rep(0, totaldur_min*60/ws3) # initialize vector fo 1 hour
+    boutdur = boutdur * 60/ws3
+    boutdur_detect = boutdur_detect * 60/ws3
+    gaps = boutdur_detect * (1-boutcriter)
+    bout = rep(1, boutdur)
+    # random position for gaps without gaps > 1min
+    gap1min = 60/ws3 + 1
+    longest_gap = gap1min + 1
+    while (longest_gap >= gap1min) {
+      zeroes = sort(sample(x = 2:(length(bout) - 1), size = gaps))
+      temp <- cumsum(c(1, diff(zeroes) - 1))
+      temp2 <- rle(temp)
+      longest_gap_tmp = zeroes[which(temp == with(temp2, values[which.max(lengths)]))]
+      longest_gap = length(longest_gap_tmp)
+      # if longest_gap <= 1min, then the random zeroes generated are fine and this breaks the loop
+    }
+    bout[zeroes] = 0
+    # random position in x to place the bouts
+    sep_between_bouts = boutdur + gap1min #bouts need to be separated at least for more than 1 min
+    sep = sep_between_bouts - 1
+    while (sep < sep_between_bouts) {
+      where = sort(sample(x = 1:(length(x) - boutdur), size = nBouts))
+      if (length(where) < 2) sep = sep_between_bouts + 1 # if only one bout, any position is good
+      if (length(where) >= 2) {
+        sep_bouts = diff(where)
+        sep = min(sep_bouts)
+      }
+    }
+    # impute bouts in x
+    for (start_i in where) x[start_i:(start_i + boutdur - 1)] = bout
+    
+    return(x)
+  }
   
+  # -------------
+  # Metric 6 - random tests with variable configurations
+  # -------------
+  for (i in 1:500) {
+    # epoch lenght will be of 5, 15, or 60 seconds
+    ws3 = c(5, 15, 60)[sample(x = 1:3, size = 1)]
+    
+    # duration of the bout to be created with dummybout()
+    # from 5 to 30 min in 5-min increments
+    boutdur = c(5, 10, 15, 20, 25, 30)[sample(x = 1:6, size = 1)]
+    
+    # boutdur in minutes to detect bouts with g.getbout 
+    # (this should always be lower than boutdur)
+    boutdur_detect = 60
+    while (boutdur_detect > boutdur) {
+      boutdur_detect = c(5, 10, 15, 25, 30)[sample(x = 1:5, size = 1)]
+    }
+    
+    # boutcriter, one of 0.8, 0.9, 1
+    boutcriter = c(0.8, 0.9, 1)[sample(x = 1:3, size = 1)]
+    
+    # number of bouts to be created
+    nBouts = c(1, 2, 3)[sample(x = 1:3, size = 1)]
+    
+    # dummy bout to test the function, we will run a vector of 24 hours with 
+    # randomly allocated nBouts
+    x = dummybout(totaldur_min = 1440, ws3 = ws3, boutdur = boutdur, boutdur_detect = boutdur_detect, 
+                  boutcriter = boutcriter, nBouts = nBouts)
+    
+    # bout detection with g.getbout
+    out = g.getbout(x = x, boutduration = boutdur_detect*60/ws3, boutcriter = boutcriter, 
```

Comment:
If I understand correctly, this unit test does not test:
- whether a bout is correctly missed when boutcriter is higher than the actual gaps in the bout?
- when bout length is lower than boutduration

It would be good if we had a unit test for all these scenarios.

Reply:
Also I now see you removed the metric 6 unit test:

```
# Metric 6 - ability to detect bouts with grace periods
  xtest = c(rep(0,15), rep(1,6), 0, rep(1,10),rep(0,20))
  bm6 = g.getbout(x=xtest,boutduration = 10, boutcriter=0.9,
                  closedbout=FALSE,bout.metric=6,ws3=60)
  expect_that(sum(bm6$x),equals(17))
```

I am now turning this on again and it fails for the revised g.getbout function.